source/material_model/melt_global.cc

```diff
@@ -240,7 +240,7 @@ namespace aspect
               const double delta_temp = in.temperature[i]-this->get_adiabatic_conditions().temperature(in.position[i]);
               visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/this->get_adiabatic_conditions().temperature(in.position[i])),1e4),1e-4);
             }
-          else
+          else if (thermal_viscosity_exponent != 0.0)
```

Comment:
Why do you check for `thermal_viscosity_exponent` and not for `reference_T`?

Reply:
We initialize the temperature-dependence with `visc_temperature_dependence = 1.0`, which is the value that corresponds to no temperature-dependence of the viscosity (because `out.viscosities[i] *= visc_temperature_dependence`). 

Accordingly, if the thermal viscosity exponent is zero (meaning we don't want any temperature-dependence), we can just leave visc_temperature_dependence at its initial value. This also has the advantage that we never get to the case where we divide by `reference_T`, so we don't have to deal with adding another if statement. Logically, I found this easier to read. 

Technically, it probably wouldn't make a difference to check for `reference_T` instead, because there is an Assert in parse_parameters that makes sure that thermal_viscosity_exponent is zero whenever reference_T is zero. So if you want me to change it, I can do that.